common/Geometry2d/Rect.cpp

```diff
@@ -18,6 +18,111 @@ bool Rect::intersects(const Rect& other) const {
     }
 }
 
+int Rect::CohenSutherlandCode(const Point& other) const{
+    const int INSIDE = 0; // 0000
+    const int LEFT = 1;   // 0001
+    const int RIGHT = 2;  // 0010
+    const int BOTTOM = 4; // 0100
+    const int TOP = 8;    // 1000
```

Comment:
You can use `0x01` to make this a hex literal or `1<<2` if you want to specify it's a bitwise field. Either of those makes it much clearer that you are using them in their binary form

Reply:
using hex literals